File: source/yeon/6주차/code/1449수리공항승.cpp
// 백준 1449 수리공 항승
// cpp solved by Greedy algorithm
#include <iostream>
#include <algorithm>
#define MAX 1000

int main(void){
    int N, L, ans = 0;
    int arr[MAX];
    scanf("%d %d",&N,&L);
    for(int i = 0; i < N; i++){
        scanf("%d",&arr[i]);
    }
    std::sort(arr, arr+N);
    float tape_range = 0;
    for(int i = 0; i < N; i++){
        if(arr[i] + 0.5 > tape_range){
            ans ++;
            tape_range = arr[i] - 0.5 + L;
        }
    }
    printf("%d",ans);
    return 0;
}

File: source/yeon/6주차/code/1449수리공항승.py
# 백준 1449 수리공 항승
# python solved by Greedy algorithm
N,L = map(int,input().split())
arr = list(map(int,input().split()))

cnt, tape_range = 0, 0
arr.sort()

# put tape first and set tape_range 
# and check arr element if its in range 
# if they are out of range, put another tape and set again
for i in range(0, len(arr)):
    if arr[i] + 0.5 > tape_range :
        cnt += 1
        tape_range = arr[i] - 0.5 + L

print(cnt)
